### PR TITLE
ENT-4519:  Added playbook, role and docker file to setup/test tableau server.

### PR DESCRIPTION
**Pull request by saleem-latif**
Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [ ] A DevOps team member has approved the PR if it is code shared across multiple services and you don't own all of the services.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [ ] If you are updating a secure value rather than an internal one, file a DEVOPS ticket with details.
    - [ ] Add an entry to the CHANGELOG.
  - [ ] If you are making a complicated change, have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?  Adding a new variable does not require the full list (although testing on a sandbox is a great idea to ensure it links with your downstream code changes).
  - [ ] Think about how this change will affect Open edX operators.  Have you updated the wiki page for the next Open edX release?

### CODE REVIEW

File: docker/build/tableau/ansible_overrides.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+TABLEAU_ADMIN_USER: tableau
+TABLEAU_ADMIN_PASSWORD: 'password'
+TABLEAU_SERVER_ADMIN_USER: "tableau-admin"
+TABLEAU_SERVER_ADMIN_PASSWORD: "password"
+
+TABLEAU_REGISTRATION_CONFIG_USER_FIRST_NAME: "Saleem"
+TABLEAU_REGISTRATION_CONFIG_USER_LAST_NAME: "Latif"
+TABLEAU_REGISTRATION_CONFIG_USER_TITLE: "Software Engineer"
+TABLEAU_REGISTRATION_CONFIG_USER_EMAIL: "salee@edx.org"
```
Comment (moconnell1453) [inline docker/build/tableau/ansible_overrides.yml]: address is missing an 'm'

File: docker/build/tableau/ansible_overrides.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+---
+TABLEAU_ADMIN_USER: tableau
```
Comment (moconnell1453) [inline docker/build/tableau/ansible_overrides.yml:2]: We've been using 'edx' and 'edx-admin' for the user names in prod. We should be consistent in usage.

File: docker/build/tableau/ansible_overrides.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+TABLEAU_ADMIN_PASSWORD: 'password'
+TABLEAU_SERVER_ADMIN_USER: "tableau-admin"
+TABLEAU_SERVER_ADMIN_PASSWORD: "password"
```
Comment (moconnell1453) [inline docker/build/tableau/ansible_overrides.yml:5]: These passwords will need to be read from Vault. We can't commit passwords to this repo.

File: playbooks/roles/tableau/defaults/main.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+#
+
+tableau_data_dir: "{{ COMMON_DATA_DIR }}/tableau"
+tableau_app_dir: "{{ COMMON_APP_DIR }}/tableau"
+tableau_log_dir: "{{ COMMON_LOG_DIR }}/tableau"
+tableau_code_dir: "{{ tableau_app_dir }}/tableau"
+tableau_installer_dir: "{{ tableau_code_dir }}/linux/automated-installer/"
+
+TABLEAU_ADMIN_USER: tableau
+TABLEAU_ADMIN_PASSWORD: 'tableau001-pass'
```
Comment (moconnell1453) [inline playbooks/roles/tableau/defaults/main.yml:19]: Passwords need to come from Vault.

File: playbooks/roles/tableau/defaults/main.yml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+tableau_user: "{{ TABLEAU_ADMIN_USER }}"
+tableau_user_createhome: 'yes'
+tableau_user_shell: '/bin/false'
+
+tableau_repo: "https://github.com/tableau/server-install-script-samples.git"
+tableau_version: 'master'
+
+tableau_server_deb_pkg: 'https://downloads.tableau.com/tssoftware/tableau-server-2020-4-1_amd64.deb'
```
Comment (moconnell1453) [inline playbooks/roles/tableau/defaults/main.yml]: This version is from January. version 2020-4-4 is the latest for this major version, but there is also 2021-1-1 available as well.

**Review by moconnell1453 — CHANGES_REQUESTED** [review]
I will check with SRE to determine how we can safely retrieve passwords from Vault. Also, can this be targeted at stage? I'd like to test there or in playground, so we don't interfere with the current server in prod.

**Comment by saleem-latif** [comment]
@moconnell1453 Everything in caps is supposed to be overridden, I have only placed these values as placeholder/test values. I was thinking to use remote config to override these values. I will discuss with SRE to get documentation for doing that.

**Comment by moconnell1453** [comment]
Thanks, @saleem-latif. I've created an SRE ticket to track their collaboration with us on this issue. https://openedx.atlassian.net/servicedesk/customer/portal/15/DOS-2376?created=true

File: playbooks/roles/tableau/defaults/main.yml (hunk 5 above)
Comment (saleem-latif) [reply playbooks/roles/tableau/defaults/main.yml]: @moconnell1453 I could not find the URL of the debian package for the release you mentioned, can you share the link?

Comment (moconnell1453) [reply playbooks/roles/tableau/defaults/main.yml]: Sure. I got these urls from this page: https://www.tableau.com/support/releases/server
Here are the latest releases (May 18th, 2021)
2021.1.2 - https://downloads.tableau.com/esdalt/2021.1.2/tableau-server-2021-1-2_amd64.deb
2020.4.5 - https://downloads.tableau.com/esdalt/2020.4.5/tableau-server-2020-4-5_amd64.deb

**Comment by saleem-latif** [comment]
@moconnell1453 I have incorporated your feedback changes, please take another look. 
Regarding your feedback regarding passwords, username and password will be overridden via edx-internal PR.

File: docker/build/tableau/ansible_overrides.yml (hunk 3 above)
Comment (moconnell1453) [reply docker/build/tableau/ansible_overrides.yml:5]: Based on our discussions with SRE, you can send the credentials to either Adam or Nadeem through LastPass, and they will add them to edx-internal: https://openedx.atlassian.net/browse/PSRE-815

**Review by moconnell1453 — APPROVED** [review]

**Pull request merged**